File: 1364-tuple-with-same-product/tuple-with-same-product.java
class Solution {
    public int tupleSameProduct(int[] nums) {
        HashMap<Integer,Integer>mp=new HashMap<>();
        int res=0;
        int n=nums.length;
        for(int i=0;i<n-1;i++){
            for(int j=i+1;j<n;j++){
                int prd=nums[i]*nums[j];
                if(mp.containsKey(prd)){
                    res+=mp.get(prd)*8;
                    mp.put(prd,mp.getOrDefault(prd,0)+1);
                }
                else{
                    mp.put(prd,1);
                }
            }

        }
        return res;
    }
}

File: 1364-tuple-with-same-product/tuple-with-same-product.py
class Solution(object):
    def tupleSameProduct(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        mp=Counter()
        l=len(nums)
        res=0
        for i in range(l-1):
            for j in range(i+1,l):
                prd=nums[i]*nums[j]
                res+=mp[prd]*8
                mp[prd]+=1
        return res

        